packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/BuddyBridge.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Copyright 2021, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pragma solidity ^0.6.11;
+
+import "arb-bridge-eth/contracts/bridge/interfaces/IInbox.sol";
+import "arb-bridge-eth/contracts/bridge/interfaces/IOutbox.sol";
+import "arb-bridge-eth/contracts/bridge/interfaces/IBridge.sol";
+
+import "../arbitrum/BuddyDeployer.sol";
+
+// contracts that want to have buddies should inherit from this
+abstract contract BuddyContract {
+    bool public connected;
+    L2Deployer public l2Deployer;
+    IInbox public inbox;
+    bytes32 codeHash;
+
+    constructor(address _inbox, address _l2Deployer) public {
+        connected = false;
+        inbox = IInbox(_inbox);
+        l2Deployer = L2Deployer(_l2Deployer);
+    }
+
+    function initiateBuddyDeploy(
```

Comment:
We might want to stop this from being called multiple times with different deploy code